net/respondd/src/respondd.c

```diff
@@ -165,6 +181,56 @@ static const struct respondd_provider_info * get_providers(const char *filename)
 	return ret;
 }
 
+bool schedule_push_request(struct request_schedule *s, struct request_task *new_task) {
+	if (s->length >= SCHEDULE_LEN)
+		// schedule is full
+		return false;
+
+	// insert into sorted list
+	struct request_task **pos;
+	for (pos = &s->list_head; *pos; pos = &((*pos)->next)) {
+		if ((*pos)->scheduled_time > new_task->scheduled_time)
+			break;
+	}
+	// insert before *pos
+	new_task->next = *pos;
+	*pos = new_task;
+
+	s->length++;
+	return true;
+}
+
+int64_t schedule_idle_time(struct request_schedule *s) {
+	if (!s->list_head)
+		// nothing to do yet (0 = infinite time)
+		return 0;
+
+	update_time();
```

Comment:
I don't think the `update_time()` here is necessary. The only part of respondd that will ever block is the recvmsg, after which the time is updated.